### PR TITLE
Add methods SetAsync/UpdateLinkAsync/DeleteLinkAsync to the Fetcher Classes

### PR DESCRIPTION
**Pull request by Sridhar-MS**
DeleteLinkAsync
 - It deletes the link between an EntityType and its single-valued navigation property.

UpdateLinkAsync
 - It updates/modifies the link between an EntityType and its single-valued navigation property.

SetAsync
 - Its sets/creates a new entity for the navigation property of an entity type.

Note: The explicit interface navigation properties are not yet removed from the data/concrete classes. It will be done when all the appropriate methods for the collection fetchers.

### CODE REVIEW

File: src/Writers/Vipr.Writer.CSharp.Lite/SourceCodeGenerator.cs [hunk 1]
```diff
@@ -1143,7 +1170,7 @@ private void WriteObsoletedPropertySet(ObsoletedProperty property)
             _("set");
             using (_builder.IndentBraced)
             {
-                _("{0} = value;", property.UpdatedName);
+                 _("{0} = value;", property.UpdatedName);
```
Comment (piotrpMSFT) [inline src/Writers/Vipr.Writer.CSharp.Lite/SourceCodeGenerator.cs]: spacing?

File: src/Writers/Vipr.Writer.CSharp.Lite/SourceCodeGenerator.cs [hunk 2]
```diff
@@ -1152,7 +1179,7 @@ private void WriteObsoletedPropertyGet(ObsoletedProperty property)
             _("get");
             using (_builder.IndentBraced)
             {
-                _("return {0};", property.UpdatedName);
```
Comment (piotrpMSFT) [inline src/Writers/Vipr.Writer.CSharp.Lite/SourceCodeGenerator.cs]: spacing?

File: test/CSharpLiteWriterUnitTests/Given_an_OdcmClass_Entity_Fetcher_DeleteLinkAsync_Method.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+            using (_mockedService = new MockService()
+                    .SetupPostEntity(TargetEntity, entityKeyValues))
+            {
+                var context = _mockedService.GetDefaultContext(Model);
+                var fetcher = context.CreateFetcher(_navPropertyFetcherType, propertyPath);
+                var sourceInstance = context.CreateConcrete(ConcreteType);
+
+                //delay save when deleting the link
+                fetcher.InvokeMethod<Task>("DeleteLinkAsync", new object[] { sourceInstance, true }).Wait();
```
Comment (piotrpMSFT) [inline test/CSharpLiteWriterUnitTests/Given_an_OdcmClass_Entity_Fetcher_DeleteLinkAsync_Method.cs]: should we make sure that the batch later does execute the delete?

**Comment by piotrpMSFT** [comment]
Can you add some 'structural' tests that show that these methods are being emitted on the right objects in the right places? E.G. Given an OdcmClassEntity  it should have an update/delete/set method. This will be useful when we start turning stuff off with capabilities.

Otherwise, LGTM.

File: test/CSharpLiteWriterUnitTests/Given_an_OdcmClass_Entity_Fetcher_DeleteLinkAsync_Method.cs (hunk 3 above)
Comment (Sridhar-MS) [inline test/CSharpLiteWriterUnitTests/Given_an_OdcmClass_Entity_Fetcher_DeleteLinkAsync_Method.cs]: There is already a test for 'SaveChangesAsync' method which tests this scenario. I believe that test is sufficient enough.

Test:
Given_an_OdcmClass_Entity_Fetcher_SaveChangesAsync_Method.It_updates_and_delay_saves_an_entity_from_its_own_path

Comment (piotrpMSFT) [inline test/CSharpLiteWriterUnitTests/Given_an_OdcmClass_Entity_Fetcher_DeleteLinkAsync_Method.cs]: I'm not comfortable with this resolution. We are building debt and and a false sense of security. We can now refactor the delete code to have it no-op when a user attempts to batch delete. We will only learn that there is an issue when a user experiences frustration.

This is an easy enhancement to add and we should do it.

**Pull request merged**